src/components/BiDiLog.js

```diff
@@ -1,14 +1,20 @@
 import React from "react";
 import BiDiLogEntry from "./BiDiLogEntry";
+import BiDCommandInput from "./BiDCommandInput";
 
 import "./BiDiLog.css";
 
-const BiDiLog = ({ log }) => {
+const BiDiLog = ({ bidiCommand, log, onBidiCommandChange, sendCommand }) => {
   return (
     <div className="bidi-log">
       <ul className="bidi-log__list">
-        {log.map((entry) => (
-          <BiDiLogEntry entry={entry} key={entry.message}/>
+        <BiDCommandInput
+          onInputChange={onBidiCommandChange}
+          onSubmit={sendCommand}
+          value={bidiCommand}
+        />
+        {log.toReversed().map((entry) => (
+          <BiDiLogEntry entry={entry} key={entry.message} />
```

Comment:
Not related to your change, but in my BiDi logs I had twice the message `WebSocket closed` and therefore they both had the same key. Because of this each re-render of the component leads to a react error, and since we re-render on every keystroke (due to the state update for inputChange), this makes the console logs hard to use.